lib/libspl/include/assert.h

```diff
@@ -42,6 +42,17 @@ __assert_c99(const char *expr, const char *file, int line, const char *func)
 }
 #endif  /* __assert_c99 */
 
+#ifndef __assert
```

Comment:
it looks like we cannot target __assert.
@behlendorf  because __assert is glibc specific, can we change this to #ifndef __GLIBC__?